publisher-sdk/src/androidTest/java/com/criteo/publisher/integration/DfpHeaderBiddingFunctionalTest.java

```diff
@@ -672,11 +691,34 @@ public void onAdLoaded() {
       }
 
       @Override
-      public void onAdFailedToLoad(int i) {
+      public void onAdFailedToLoad(LoadAdError loadAdError) {
+        LoggerFactory.getLogger(DfpHeaderBiddingFunctionalTest.class)
+            .debug("onAdFailedToLoad for reason %d. "
+                    + "See: https://developers.google.com/android/reference/com/google/android/gms/ads/doubleclick/PublisherAdRequest",
+                loadAdError);
+        isLoaded.countDown();
+      }
+    }
+
+    public class SyncAdManagerInterstitialAdLoadCallback extends AdManagerInterstitialAdLoadCallback {
+      private AdManagerInterstitialAd _adManagerInterstitialAd;
+
+      public AdManagerInterstitialAd getAdManagerInterstitialAd() {
+        return _adManagerInterstitialAd;
+      }
+
+      @Override
+      public void onAdLoaded(@NonNull AdManagerInterstitialAd adManagerInterstitialAd) {
+        _adManagerInterstitialAd = adManagerInterstitialAd;
```

Comment:
If this side effect bother you, maybe could you rename the attribute to reflect what it is: `lastInterstitialAd` ?

Reply:
Ah good suggestion ! done